source/Examples/SmartCache/Main.cpp

```diff
@@ -56,40 +76,82 @@ main(
     //
     ISharedChannel& feedbackChannel = mlosContext.FeedbackChannel();
 
+    // This background thread uses a lambda to monitor the feedback channel for
+    // new messages and process them using the callbacks registered for each
+    // message type in the global dispatch table.
+    //
     std::future<bool> feedbackChannelReader = std::async(
```

Comment:
how is the return value used? just do indicate that the thread finished?

Reply:
It's not at the moment and ideally the compiler should complain about that (#22).